### Import Libraries

In [1]:
import pandas as pd
from pathlib import Path
from sklearn.model_selection import train_test_split

### Config

In [2]:
# Constants
RANDOM_STATE = 42
TARGET_COL = "label"
TEXT_COL = "text"
TEST_SIZE = 0.25

# Base path
BASE_DIR = Path.cwd().resolve().parents[1]

# Paths
DATA_DIR = BASE_DIR / "data"
PROCESSED_DIR = DATA_DIR / "processed"
PROCESSED_DATA_PATH = PROCESSED_DIR / "cleaned_sms.csv"

### Load Dataset

In [3]:
df = pd.read_csv(PROCESSED_DATA_PATH)
print("Shape:", df.shape)
print("\nData Types:\n", df.dtypes)

Shape: (5157, 3)

Data Types:
 label       object
raw_text    object
text        object
dtype: object


In [4]:
# Preview
df.head(5)

,label,raw_text,text
0,ham,"Go until jurong point, crazy.. Available only ...","Go until jurong point, crazy.. Available only ..."
1,ham,Ok lar... Joking wif u oni...,Ok lar... Joking wif you oni...
2,spam,Free entry in 2 a wkly comp to win FA Cup fina...,Free entry in 2 a wkly comp to win FA Cup fina...
3,ham,U dun say so early hor... U c already then say...,YOU dun say so early hor... YOU c already then...
4,ham,"Nah I don't think he goes to usf, he lives aro...","Nah I do not think he goes to usf, he lives ar..."


In [5]:
# Define Features and Target
X = df[TEXT_COL]
y = df[TARGET_COL]

In [6]:
# Class distribution
print(y.value_counts(normalize=True).rename("ratio"))

label
ham     0.875703
spam    0.124297
Name: ratio, dtype: float64


**Observation**
- Processed dataset loaded successfully with 5157 rows and 3 columns
- `text` column will be used as input features, and `label` as target
- Class distribution remains imbalanced (~87.6% ham), consistent with preprocessing stage

### Train/Test split

In [7]:
X_train, X_test, y_train, y_test = train_test_split(X, y, 
                                                    test_size=TEST_SIZE, 
                                                    random_state=RANDOM_STATE,
                                                    stratify=y)

In [8]:
# Shape check
print("X_train:", X_train.shape)
print("X_test :", X_test.shape)
print("y_train:", y_train.shape)
print("y_test :", y_test.shape)

X_train: (3867,)
X_test : (1290,)
y_train: (3867,)
y_test : (1290,)


In [9]:
# Distribution check
print("\nTrain distribution:")
print(y_train.value_counts(normalize=True).rename("ratio"))

print("\nTest distribution:")
print(y_test.value_counts(normalize=True).rename("ratio"))


Train distribution:
label
ham     0.875614
spam    0.124386
Name: ratio, dtype: float64

Test distribution:
label
ham     0.875969
spam    0.124031
Name: ratio, dtype: float64


**Observation**
- Dataset successfully split into train and test sets (75/25 split)
- Train set contains 3867 samples and test set contains 1290 samples
- Class distribution is preserved across train and test sets (≈87.6% ham, ≈12.4% spam), confirming correct stratification
- This ensures balanced representation of classes and supports reliable model evaluation